pkg/asset/deploy/deploy_iso.go

```diff
@@ -0,0 +1,165 @@
+package deploy
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/appliance/pkg/asset/config"
+	"github.com/openshift/appliance/pkg/asset/ignition"
+	"github.com/openshift/appliance/pkg/asset/recovery"
+	"github.com/openshift/appliance/pkg/consts"
+	"github.com/openshift/appliance/pkg/coreos"
+	"github.com/openshift/appliance/pkg/fileutil"
+	"github.com/openshift/appliance/pkg/log"
+	"github.com/openshift/appliance/pkg/skopeo"
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/sirupsen/logrus"
+)
+
+type DeployISO struct {
+	File *asset.File
+}
+
+var _ asset.Asset = (*DeployISO)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *DeployISO) Name() string {
+	return "Deployment ISO"
+}
+
+// Dependencies returns dependencies used by the asset.
+func (i *DeployISO) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&config.EnvConfig{},
+		&config.ApplianceConfig{},
+		&ignition.DeployIgnition{},
+		&recovery.BaseISO{},
+	}
+}
+
+// Generate the base ISO.
+func (i *DeployISO) Generate(dependencies asset.Parents) error {
+	envConfig := &config.EnvConfig{}
+	applianceConfig := &config.ApplianceConfig{}
+	baseISO := &recovery.BaseISO{}
+	deployIgnition := &ignition.DeployIgnition{}
+
+	dependencies.Get(envConfig, applianceConfig, baseISO, deployIgnition)
+
+	// Search for deployment ISO in cache dir
+	if fileName := envConfig.FindInAssets(consts.DeployIsoName); fileName != "" {
+		logrus.Info("Configuring appliance deployment ISO")
+		i.File = &asset.File{Filename: fileName}
+	} else if err := i.buildDeploymentIso(envConfig, applianceConfig); err != nil {
+		return err
+	}
+
+	// Embed ignition in ISO
+	coreOSConfig := coreos.CoreOSConfig{
+		ApplianceConfig: applianceConfig,
+		EnvConfig:       envConfig,
+	}
+	c := coreos.NewCoreOS(coreOSConfig)
+	ignitionBytes, err := json.Marshal(deployIgnition.Config)
+	if err != nil {
+		logrus.Errorf("Failed to marshal recovery ignition to json: %s", err.Error())
+		return err
+	}
+	deployIsoFileName := filepath.Join(envConfig.AssetsDir, consts.DeployIsoName)
+	if err = c.EmbedIgnition(ignitionBytes, deployIsoFileName); err != nil {
+		logrus.Errorf("Failed to embed ignition in recovery ISO: %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (i *DeployISO) buildDeploymentIso(envConfig *config.EnvConfig, applianceConfig *config.ApplianceConfig) error {
+	if fileName := envConfig.FindInAssets(consts.ApplianceFileName); fileName == "" {
+		logrus.Infof("The appliance.raw disk image file is missing.")
+		logrus.Infof("Run 'build' command for building the appliance disk image.")
+		logrus.Exit(0)
```

Comment:
try to see what happens if you set Exit(1), since it does fail to create the ISO in this case.

Reply:
Done